File: services/payment-service/src/main/java/io/minishop/payment/web/dto/PgChargeResponse.java
package io.minishop.payment.web.dto;

public record PgChargeResponse(
        boolean success,
        String reference,
        String reason
) {
    public static PgChargeResponse ok(String reference) {
        return new PgChargeResponse(true, reference, null);
    }

    public static PgChargeResponse fail(String reason) {
        return new PgChargeResponse(false, null, reason);
    }
}

File: services/payment-service/src/main/kotlin/io/minishop/payment/web/dto/PgChargeResponse.kt
package io.minishop.payment.web.dto

@JvmRecord
data class PgChargeResponse(
    val success: Boolean,
    val reference: String?,
    val reason: String?,
) {
    companion object {
        @JvmStatic
        fun ok(reference: String): PgChargeResponse = PgChargeResponse(true, reference, null)

        @JvmStatic
        fun fail(reason: String): PgChargeResponse = PgChargeResponse(false, null, reason)
    }
}
